glean-core/src/metrics/labeled.rs

```diff
@@ -351,6 +353,23 @@ where
     }
 }
 
+impl<T, S> TestGetLabeledValues<S> for LabeledMetric<T>
+where
+    T: AllowLabeled + TestGetValue<S>,
+    S: Any,
+{
+    fn test_get_labeled_values(&self, ping_name: Option<String>) -> HashMap<String, S> {
+        let mut out = HashMap::new();
+        let map = self.label_map.lock().unwrap();
+        map.iter().for_each(|(label, submetric)| {
+            if let Some(v) = submetric.test_get_value(ping_name.clone()) {
+                out.insert(label.clone(), v);
+            }
+        });
+        out
```

Comment:
For fun I tried to see if I could write this functionally. Not sure whether it's any more or less readable, though I do like having space to mention why we think the `unwrap()` is okay to call.

```rust
self.label_map
    .lock()
    .unwrap() // Safe unless lock's poisoned. Also: this is a test-only method: we're allowed to panic.
    .iter()
    .filter_map(|(label, submetric)| {
        submetric.test_get_value(ping_name.clone()).map(|v| (label.clone(), v))
    })
    .collect()
```

I haven't even checked to see if it would compile, but you're welcome to adapt it or drop it: your choice.

Reply:
I'm also now torn between keeping up with the piping like what you suggested or going even more imperative with a `for` on the iter. Not really sure which is more readable

Reply:
Then let's go with as-written since there's nothing unreadable about it. Just drop in a crash-safety comment on the `unwrap()`

Reply:
I'm actually going to end up committing my change from what's here to the more imperative `for`